include/nbl/video/utilities/CComputeBlit.h

```diff
@@ -0,0 +1,900 @@
+		const asset::ISampler::E_TEXTURE_CLAMP wrapV = asset::ISampler::ETC_CLAMP_TO_EDGE,
+		const asset::ISampler::E_TEXTURE_CLAMP wrapW = asset::ISampler::ETC_CLAMP_TO_EDGE,
+		const asset::ISampler::E_TEXTURE_BORDER_COLOR borderColor = asset::ISampler::ETBC_FLOAT_OPAQUE_BLACK)
+	{
+		constexpr auto MAX_DESCRIPTOR_COUNT = 3;
+
+		auto updateDS = [this, coverageAdjustmentScratchBuffer](video::IGPUDescriptorSet* ds, video::IGPUDescriptorSet::SDescriptorInfo* infos) -> bool
+		{
+			const auto& bindings = ds->getLayout()->getBindings();
+			if ((bindings.size() == 3) && !coverageAdjustmentScratchBuffer)
+				return false;
+
+			video::IGPUDescriptorSet::SWriteDescriptorSet writes[MAX_DESCRIPTOR_COUNT] = {};
+
+			for (auto i = 0; i < bindings.size(); ++i)
+			{
+				writes[i].dstSet = ds;
+				writes[i].binding = i;
+				writes[i].arrayElement = 0u;
+				writes[i].count = 1u;
+				writes[i].info = &infos[i];
+				writes[i].descriptorType = bindings.begin()[i].type;
+			}
+
+			m_device->updateDescriptorSets(bindings.size(), writes, 0u, nullptr);
+
+			return true;
+		};
+
+		if (blitDS)
+		{
+			if (!inImageView || !outImageView)
+				return false;
+
+			video::IGPUDescriptorSet::SDescriptorInfo infos[MAX_DESCRIPTOR_COUNT] = {};
+
+			if (!samplers[wrapU][wrapV][wrapW][borderColor])
+			{
+				video::IGPUSampler::SParams params = {};
+				params.TextureWrapU = wrapU;
+				params.TextureWrapV = wrapV;
+				params.TextureWrapW = wrapW;
+				params.BorderColor = borderColor;
+				params.MinFilter = asset::ISampler::ETF_NEAREST;
+				params.MaxFilter = asset::ISampler::ETF_NEAREST;
+				params.MipmapMode = asset::ISampler::ESMM_NEAREST;
+				params.AnisotropicFilter = 0u;
+				params.CompareEnable = 0u;
+				params.CompareFunc = asset::ISampler::ECO_ALWAYS;
+
+				samplers[wrapU][wrapV][wrapW][borderColor] = m_device->createSampler(params);
+				if (!samplers[wrapU][wrapV][wrapW][borderColor])
+					return false;
+			}
+			
+			infos[0].desc = inImageView;
+			infos[0].image.imageLayout = asset::EIL_SHADER_READ_ONLY_OPTIMAL;
+			infos[0].image.sampler = samplers[wrapU][wrapV][wrapW][borderColor];
+
+			infos[1].desc = outImageView;
+			infos[1].image.imageLayout = asset::EIL_GENERAL;
+			infos[1].image.sampler = nullptr;
+
+			if (coverageAdjustmentScratchBuffer)
+			{
+				infos[2].desc = coverageAdjustmentScratchBuffer;
+				infos[2].buffer.offset = 0;
+				infos[2].buffer.size = coverageAdjustmentScratchBuffer->getSize();
+			}
+
+			if (!updateDS(blitDS, infos))
+				return false;
+		}
+
+		if (kernelWeightsDS)
+		{
+			video::IGPUDescriptorSet::SDescriptorInfo info = {};
+			info.desc = kernelWeightsUTB;
+			info.buffer.offset = 0ull;
+			info.buffer.size = kernelWeightsUTB->getUnderlyingBuffer()->getSize();
+
+			if (!updateDS(kernelWeightsDS, &info))
+				return false;
+		}
+
+		return true;
+	}
+
+	template <typename KernelX, typename KernelY, typename KernelZ>
+	inline void blit(
```

Comment:
please write a comment that the user is responsible for the memory barriers between previous writes of the `inImage` and the first dispatch and future reads of `outImage` and last dispatch

Reply:
Resolved.